ecosystem/typescript/sdk_v2/src/crypto/ed25519.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { Deserializer, Serializer } from "../bcs";
+import { Hex } from "../core";
+import { HexInput } from "../types";
+
+export class Ed25519PublicKey {
+  static readonly LENGTH: number = 32;
+
+  readonly value: Hex;
+
+  constructor(value: HexInput) {
+    if (value.length !== Ed25519PublicKey.LENGTH) {
+      throw new Error(`Ed25519PublicKey length should be ${Ed25519PublicKey.LENGTH}`);
+    }
+    this.value = Hex.fromHexInput({ hexInput: value });
```

Comment:
Not sure if its just me. I found the obj for a single arg is not very user friendly

Reply:
Yeah not a fan. I'd be much happier with no object for single arg, object args for 2+ args, personally